cns/restserver/internalapi.go

```diff
@@ -151,6 +152,37 @@ func (service *HTTPRestService) SyncNodeStatus(dncEP, infraVnet, nodeID string,
 	return
 }
 
+// SyncHostNCVersion will check NC version from NMAgent and save it as host version in container status.
+// If NMAgent updated, CNS will refresh the pending programming IP status.
+func (service *HTTPRestService) SyncHostNCVersion(channelMode string) {
+	for i, containerstatus := range service.state.ContainerStatus {
+		// Will open a separate PR to convert all the NC version related variable to int. Change from string to int is a pain.
+		hostVersion, err := strconv.Atoi(containerstatus.HostVersion)
+		if err != nil {
+			log.Errorf("Received err when chagne containerstatus.HostVersion %s to int, err msg %v", containerstatus.HostVersion, err)
+			return
+		}
+		vmVersion, err := strconv.Atoi(containerstatus.VMVersion)
+		if err != nil {
+			log.Errorf("Received err when chagne containerstatus.VMVersion %s to int, err msg %v", containerstatus.VMVersion, err)
+			return
+		}
+		// host NC version is the NC version from NMAgent, if it's already keep up with NC version exist in VM, no update needed.
+		if hostVersion >= vmVersion {
+			continue
+		} else {
+			newHostNCVersion := service.imdsClient.GetNetworkContainerInfoFromHostWithoutToken()
+			service.Lock()
+			if channelMode == cns.KubernetesCRD {
```

Comment:
Instead can you check if this NC has secondaryIps?

Reply:
Check whether NC has secondarIps is taken care by the function inside this if statement. I tested it, if no secondaryIps, loop through secondaryIps will return directly.
If we only use SyncHostNCVersion for CRD mode for now, I can remove this check.
If other scenario, e.g. managed DNC want to extend our SyncHostNCVersion function, we can add channelMode back.